src/openrct2/GameState.cpp

```diff
@@ -277,33 +289,51 @@ void GameState::UpdateLogic()
 
     date_update();
     _date = Date(static_cast<uint32_t>(gDateMonthsElapsed), gDateMonthTicks);
+    report_time(LogicTimePart::Date);
 
     scenario_update();
+    report_time(LogicTimePart::Scenario);
     climate_update();
+    report_time(LogicTimePart::Climate);
     map_update_tiles();
+    report_time(LogicTimePart::MapTiles);
     // Temporarily remove provisional paths to prevent peep from interacting with them
```

Comment:
I wonder if it would be better to do something like:

```c++
void GameState::UpdateLogic(LogicTimings* timings)
{
    for (...)
        RunStep(step);
}

void GameLogic::RunStep(LogicTimePart step)
{
    switch(step)
    {
        case LogicTimePart::Network: ...
        case ...
    }
    report_time(step);
}
```

This would force us to write more enums and to divide the code better, but also would be a little bit of less repetition? It would also be clearer which steps are and aren't being measured?

Reply:
I'm not quite fond of this, even with the code moved out to `switch`, there's little benefit - same code, just in different place. On top of that, the `report_time` is a lambda with captures, which could cause problems if it were to be passed to the other function - or possibly incur penalty of recreating it every time we call the `GameLogic::RunStep`

Reply:
Alright